fs-cache/src/cache.rs

```diff
@@ -0,0 +1,668 @@
+    ///
+    /// First collects metadata for all files, sorts them by modification time, and then loads as many
+    /// recent files as possible within the memory limit. Files that don't fit in memory remain only
+    /// on disk.
+    fn load_fs(&mut self) -> Result<()> {
+        if !self.path.exists() {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                "Folder does not exist".to_owned(),
+            ));
+        }
+
+        if !self.path.is_dir() {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                "Path is not a directory".to_owned(),
+            ));
+        }
+
+        // Collect metadata for all files
+        let mut file_metadata = Vec::new();
+        for entry in fs::read_dir(&self.path)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file() {
+                let key: K =
+                    extract_key_from_file_path(&self.label, &path, true)?;
+                let metadata = entry.metadata()?;
+                let modified = metadata.modified()?;
+                let size = metadata.len() as usize;
+                file_metadata.push((key, size, modified));
+            }
+        }
+
+        // Sort by modified time (most recent first)
+        file_metadata.sort_by(|a, b| b.2.cmp(&a.2));
+
+        // Clear existing cache
+        self.memory_cache.clear();
+        self.current_memory_bytes = 0;
+
+        // Load files that fit in memory
+        let mut loaded_bytes = 0;
+        let total_bytes: usize = file_metadata
+            .iter()
+            .map(|(_, size, _)| size)
+            .sum();
+
+        for (key, size, _) in file_metadata {
+            if loaded_bytes + size <= self.max_memory_bytes {
+                match self.read_from_disk(&key) {
+                    Ok(value) => {
+                        self.memory_cache
+                            .put(key, CacheEntry { value, size });
+                        loaded_bytes += size;
+                    }
+                    Err(err) => {
+                        log::warn!(
+                            "cache/{}: failed to load key={}: {}",
+                            self.label,
+                            key,
+                            err
+                        );
+                    }
+                }
+            }
+        }
+
+        self.current_memory_bytes = loaded_bytes;
+
+        log::debug!(
+            "cache/{}: loaded {}/{} bytes in memory",
+            self.label,
+            self.current_memory_bytes,
+            total_bytes
+        );
+
+        Ok(())
+    }
+
+    /// Retrieves a value from the memory cache.
+    fn fetch_from_memory(&mut self, key: &K) -> Option<V> {
+        self.memory_cache
+            .get(key)
+            .map(|entry| entry.value.clone())
+    }
```

Comment:
If we remove `CacheEntry` (essentially removing the extra field for size) and use `V` directly, we can simply call `self.memory_cache.get(key)`. This would avoid cloning during each get operation, which I think can add up.